Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_DashTeleport.cs

```diff
@@ -0,0 +1,114 @@
+namespace VRTK
+{
+    using UnityEngine;
+    using System.Collections;
+    using System;
+    using System.Linq;
+
+    public class VRTK_DashTeleport : VRTK_HeightAdjustTeleport
+    {
+
+        //public float blinkTransitionSpeed = 0.0f; //not sure how to override the default value
+        public delegate void DashAction(RaycastHit[] hit);
+        public static event DashAction OnWillDashThruObjects;
+        public static event DashAction OnDashedThruObjects;
+        public float normalLerpTime = 0.1f; //100ms for every dash above minDistanceForNormalLerp, value taken from realities (www.realities.io)
+        public float minSpeedMps = 50.0f; //they also clamp to minimum speed 50m/s to avoid sickness.
+        public float capsuleTopOffset = 0.2f;
+        public float capsuleBottomOffset = 0.5f;
+        public float capsuleRadius = 0.5f;
+
+        // Dash to first obstacle -- not working yet!
+        //public bool stopAtObstacle = false;
+
+        // This is determined by the minSpeed and the normalLerpTime, if you want to always lerp with a fixed mps speed, set the normalLerpTime to a high value
+        private float minDistanceForNormalLerp;
+        private float lerpTime = 0.1f;
+
+        protected override void Start()
+        {
+            base.Start();
+            minDistanceForNormalLerp = minSpeedMps * normalLerpTime; //default values give 5.0f
+        }
+
+        protected override void SetNewPosition(Vector3 position, Transform target)
+        {
+            Vector3 targetPosition = CheckTerrainCollision(position, target);
+            StartCoroutine(lerpToPosition(targetPosition, target));
+        }
+
+        private IEnumerator lerpToPosition(Vector3 targetPosition, Transform target)
+        {
+            base.enableTeleport = false;
+            bool gameObjectInTheWay = false;
+
+            // Find the objects we will be dashing through and broadcast them via events
+            Vector3 eyeCameraPosition = eyeCamera.transform.position;
+            Vector3 eyeCameraPositionOnGround = new Vector3(eyeCameraPosition.x, this.transform.position.y, eyeCameraPosition.z);
+            Vector3 eyeCameraRelativeToRig = eyeCameraPosition - transform.position;
+            Vector3 targetEyeCameraPosition = targetPosition + eyeCameraRelativeToRig;
+            Vector3 direction = (targetEyeCameraPosition - eyeCameraPosition).normalized;
+            Vector3 bottomPoint = eyeCameraPositionOnGround + (Vector3.up * capsuleBottomOffset) + direction;
+            Vector3 topPoint = eyeCameraPosition + (Vector3.up * capsuleTopOffset) + direction;
+            float maxDistance = Vector3.Distance(transform.position, targetPosition - direction * 0.5f);
+            RaycastHit[] allHits = Physics.CapsuleCastAll(bottomPoint, topPoint, capsuleRadius, direction, maxDistance);
+
+            // Dash to first obstacle -- not working yet!
```

Comment:
delete this stuff if it's not working, maybe move it to another local development branch so you can keep working on it in the future.